crates/re_viewport/src/viewport_blueprint.rs

```diff
@@ -184,62 +215,63 @@ impl<'a> ViewportBlueprint<'a> {
         }
     }
 
-    pub fn mark_user_interaction(&mut self) {
+    pub fn mark_user_interaction(&self, ctx: &ViewerContext<'_>) {
         if self.auto_layout {
             re_log::trace!("User edits - will no longer auto-layout");
         }
 
-        self.auto_layout = false;
-        self.auto_space_views = false;
+        self.set_auto_layout(false, ctx);
+        self.set_auto_space_views(false, ctx);
     }
 
-    pub fn add_space_view(&mut self, mut space_view: SpaceViewBlueprint) -> SpaceViewId {
-        let space_view_id = space_view.id;
+    /// Add a set of space views to the viewport.
+    /// NOTE: Calling this more than once per frame will result in lost data.
```

Comment:
a note on why that is would be useful

It's because the tree action writes all of `IncludedSpaceViews` which is a list of _all_ space views, right? Hmm if we had &mut self here we could change the state we previously grabbed from the store, but yeah I can see why you opted not to do that!